### PR TITLE
HF hub notebook

### PR DESCRIPTION
**Pull request by igor-davidyuk**
This notebook covers Hugging Face public model repo

Ticket: 120317

### CODE REVIEW

File: notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openvino as ov\n",
+    "\n",
+    "ov_model = ov.convert_model(model, example_input=dict(encoded_input))"
```
Comment (aleksandr-mokrov) [inline notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:184]: Why didn't you add a model saving to avoid converting next time?

Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:184]: did not think about it, as it is not a conversion tutorial but a model repo showcase. Will add anyway

File: notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: fixed

File: notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "compiled_model = core.compile_model(save_model_path, device.value)\n",
+    "\n",
+    "ov_encoder_input = [v for v in dict(encoded_input).values()]\n",
+    "\n",
+    "# Compiled model call is performed using the same parameters as for the original model\n",
+    "scores_ov = compiled_model(ov_encoder_input)[compiled_model.outputs[0]]\n",
```
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb]: changed subscripting

**Review by aleksandr-mokrov — APPROVED** [review]
LGFM!

File: notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb [hunk 3]
```diff
@@ -0,0 +1,463 @@
+{
```
Comment (eaidova) [inline notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>probably need to provide info about torchscript=True </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (eaidova) [inline notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p><code>ov.convert_model</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (eaidova) [inline notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>Learn more on <a href="https://huggingface.co/docs/optimum/intel/inference" target="_blank">the Hugging Face website</a>.</p><p>-&gt; Learn more in Optimum Intel documentation</p><p> </p><p>or something similar, it is important that link has clear description which info provided inside, other way it do not give trust to user and will be just skipped</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (eaidova) [inline notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>it is also important to show or mention save_pretrained method that allow to save model on disk for avoid next time conversion.</p><p> </p><p>Also good to mention that Optimum Intel accepts already converted models too and the openvino models can be downloaded directly from hf hub.</p><p>they can be found by this link <a href="https://huggingface.co/models?library=openvino&amp;sort=trending" target="_blank">https://huggingface.co/models?library=openvino&amp;sort=trending</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (eaidova) [inline notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>Stable Diffusion XL, Riffusion, Dolly v2, llm chatbot also should be here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (apaniukov) [inline notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    ov_encoder_input = [v for v in dict(encoded_input).values()]
</pre><p>Is list necessary?</p><pre class="ql-syntax" spellcheck="false">ov_encoder_input = encoded_input.data
</pre><p>Using dict seems cleaner</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>I still need to cast </p><pre class="ql-syntax" spellcheck="false">dict(encoded_input)
</pre><p>but it looks cleaner.</p><p>Done</p>

Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>done!</p>

Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>done</p>

Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>fixed</p>

Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>added</p>

Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: <p>done</p>

Comment (apaniukov) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: No need to create a new dict, `encoded_input.data` attribute is already a dict:
image

Comment (igor-davidyuk) [reply notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb:1]: done

**Review by eaidova — APPROVED** [review]

**Pull request merged**